### ISSUE TITLE
False positive FCBL_FIELD_COULD_BE_LOCAL with final field

### ISSUE THREAD

**Issue by jsotuyod** [mebigfatguy/fb-contrib#200]
When having a final field, which is initialized on the constructor (configuration, for instance, when using inversion of control), if such field is only used on a single method the field is incorrectly flagged.

Code sample showing the issue:

```java
public class LocationCallbackImpl implements LocationCallback {
    @SuppressFBWarnings(value = "FCBL_FIELD_COULD_BE_LOCAL", justification = "It cannot be made local since its configured at construction by IoC")
    private final SavedLocationStorage savedLocationStorage;

    public LocationCallbackImpl(SavedLocationStorage savedLocationStorage) {
        this.savedLocationStorage = savedLocationStorage;
    }

    @Override
    public void onLocationObtained(@NonNull Geolocation location) {
        savedLocationStorage.save(location);
    }
}
```

**Comment by mebigfatguy** [comment]
When i reimplement this as the following, i'm not seeing this problem. Not sure what would be different that would matter.

```java
public class LocationCallbackImpl {
    private final SavedLocationStorage savedLocationStorage;

    public LocationCallbackImpl(SavedLocationStorage savedLocationStorage) {
        this.savedLocationStorage = savedLocationStorage;
    }

    public void onLocationObtained(Geolocation location) {
        savedLocationStorage.save(location);
    }
}


class SavedLocationStorage {
	public void save(Geolocation location) {
	}
}

class Geolocation {
}
```java

**Comment by jsotuyod** [comment]
@mebigfatguy I'll check I'm using the latest release just in case, and let you know. Maybe this was fixed and I missed it.

**Comment by jsotuyod** [comment]
Confirmed, this was fixed sometime between 6.8.x and 7.0. Closing it.

Thanks!
